txnbuild/liquidity_pool_deposit.go

```
@@ -18,6 +20,49 @@ type LiquidityPoolDeposit struct {
 	MaxPrice        string
 }
 
+// NewLiquidityPoolDeposit creates a new LiquidityPoolDeposit operation,
+// implicitly ordering assets to generate the correct pool id. minPrice, and
+// maxPrice are in terms of a/b. Each AssetAmount is a pair of the asset with
+// the maximum amount of that asset to deposit.
+func NewLiquidityPoolDeposit(
+	sourceAccount string,
+	a, b AssetAmount,
+	minPrice,
+	maxPrice string,
+) (LiquidityPoolDeposit, error) {
+	if b.Asset.LessThan(a.Asset) {
+		a, b = b, a
+
+		// Invert the minPrice. amountA/amountB -> amountB/amountA
+		min := &big.Rat{}
+		if _, ok := min.SetString(minPrice); !ok {
+			return LiquidityPoolDeposit{}, errors.Errorf("cannot parse minPrice: %v", minPrice)
+		}
+		minPrice = min.Inv(min).FloatString(7)
+
+		// Invert the maxPrice. amountA/amountB -> amountB/amountA
+		max := &big.Rat{}
+		if _, ok := max.SetString(maxPrice); !ok {
+			return LiquidityPoolDeposit{}, errors.Errorf("cannot parse maxPrice: %v", minPrice)
+		}
+		maxPrice = max.Inv(max).FloatString(7)
```

Given there can be rounding error after the price inversion I wonder if we should simply return an error here.

---

yeah, was debating that as well...

---

Per discussion, this opens a can of worms. Let's just go back to throwing an error if the assets are out-of-order. That is the safest option, matches the JS SDK, and shouldn't happen often IRL, as the assets returned from horizon will be in-order.